.github/workflows/tests.yml

```
@@ -1,28 +1,23 @@
 name: Tests
 
-on: [push, pull_request]
+on: [push]
 
 jobs:
   tests:
-    name: Tests
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout repository and submodules
-      uses: actions/checkout@v2
-      with:
-        submodules: recursive
+      - name: Checkout repository
+        uses: actions/checkout@v3
 
-    - name: Install nix 2.3.6
-      uses: cachix/install-nix-action@v13
-      with:
-        install_url: https://releases.nixos.org/nix/nix-2.3.6/install
-        nix_path: nixpkgs=channel:nixos-unstable
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
 
-    - name: Use maker and dapp cachix
-      uses: cachix/cachix-action@v10
-      with:
-        name: maker
-        extraPullNames: dapp
+      - name: Install Dependencies
+        run: git submodule update --init --recursive
 
-    - name: Run tests
-      run: nix-shell --pure --argstr url ${{ secrets.ETH_RPC_URL }} --run 'dapp test -v --rpc'
+      - name: Run tests
+        run: make test
+        env:
+          ETH_RPC_URL: ${{ secrets.ETH_RPC_URL }}
```

sharing to try and consider together with the forge tests
```suggestion
          ETH_RPC_URL: ${{ secrets.ETH_RPC_URL }}
          
        - name: Coverage
          run: forge coverage
          ETH_RPC_URL: ${{ secrets.ETH_RPC_URL }}
```

---

pending team discussion on best approach first (if a condition for failure should be explored instead of just informational output), resolving for now.